File: JavaScript/packages/recognizers-choice/src/choice/choiceRecognizer.ts
import { Recognizer, IModel, Culture, ModelResult } from "@microsoft/recognizers-text";
import { BooleanModel } from "./models";
import { BooleanExtractor } from "./extractors";
import { BooleanParser } from "./parsers";
import { ChineseBooleanExtractorConfiguration } from "./chinese/boolean";
import { DutchBooleanExtractorConfiguration } from "./dutch/boolean";
import { EnglishBooleanExtractorConfiguration } from "./english/boolean";
import { FrenchBooleanExtractorConfiguration } from "./french/boolean";
import { GermanBooleanExtractorConfiguration } from "./german/boolean";
import { JapaneseBooleanExtractorConfiguration } from "./japanese/boolean";
import { PortugueseBooleanExtractorConfiguration } from "./portuguese/boolean";
import { SpanishBooleanExtractorConfiguration } from "./spanish/boolean";


export enum ChoiceOptions {
    None = 0,
}

export function recognizeBoolean(query: string, culture: string, options: ChoiceOptions = ChoiceOptions.None,
        fallbackToDefaultCulture: boolean = true): Array<ModelResult> {
    let recognizer = new ChoiceRecognizer(culture, options);
    let model = recognizer.getBooleanModel(culture, fallbackToDefaultCulture);
    return model.parse(query);
}

export default class ChoiceRecognizer extends Recognizer<ChoiceOptions> {
    constructor(culture: string, options: ChoiceOptions = ChoiceOptions.None, lazyInitialization: boolean = false) {
        super(culture, options, lazyInitialization);
    }

    protected InitializeConfiguration() {
        //#region Chinese
        this.registerModel("BooleanModel", Culture.Chinese, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new ChineseBooleanExtractorConfiguration())
        ));
        //#endregion

        //#region Dutch
        this.registerModel("BooleanModel", Culture.Dutch, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new DutchBooleanExtractorConfiguration())
        ));
        //#endregion

        //#region English
        this.registerModel("BooleanModel", Culture.English, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new EnglishBooleanExtractorConfiguration())
        ));
        //#endregion

        //#region French
        this.registerModel("BooleanModel", Culture.French, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new FrenchBooleanExtractorConfiguration())
        ));
        //#endregion

        //#region German
        this.registerModel("BooleanModel", Culture.German, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new GermanBooleanExtractorConfiguration())
        ));
        //#endregion

        //#region Japanese
        this.registerModel("BooleanModel", Culture.Japanese, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new JapaneseBooleanExtractorConfiguration())
        ));
        //#endregion

        //#region Portuguese
        this.registerModel("BooleanModel", Culture.Portuguese, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new PortugueseBooleanExtractorConfiguration())
        ));
        //#endregion

        //#region Spanish
        this.registerModel("BooleanModel", Culture.Spanish, (options) => new BooleanModel(
            new BooleanParser(),
            new BooleanExtractor(new SpanishBooleanExtractorConfiguration())
        ));
        //#endregion
    }

    protected IsValidOptions(options: number): boolean {
        return options >= 0 && options <= ChoiceOptions.None
    }

    getBooleanModel(culture: string = null, fallbackToDefaultCulture: boolean = true): IModel {
        return this.getModel("BooleanModel", culture, fallbackToDefaultCulture);
    }
}

File: Java/libraries/recognizers-text-choice/src/main/java/com/microsoft/recognizers/text/choice/ChoiceRecognizer.java
package com.microsoft.recognizers.text.choice;

import com.microsoft.recognizers.text.Culture;
import com.microsoft.recognizers.text.IModel;
import com.microsoft.recognizers.text.ModelResult;
import com.microsoft.recognizers.text.Recognizer;
import com.microsoft.recognizers.text.choice.english.extractors.EnglishBooleanExtractorConfiguration;
import com.microsoft.recognizers.text.choice.extractors.BooleanExtractor;
import com.microsoft.recognizers.text.choice.models.BooleanModel;
import com.microsoft.recognizers.text.choice.parsers.BooleanParser;

import java.util.List;

public class ChoiceRecognizer extends Recognizer<ChoiceOptions> {

    public ChoiceRecognizer(String targetCulture, ChoiceOptions options, boolean lazyInitialization) {
        super(targetCulture, options, lazyInitialization);
    }

    public ChoiceRecognizer(String targetCulture, int options, boolean lazyInitialization) {
        this(targetCulture, ChoiceOptions.values()[options], lazyInitialization);
    }

    public ChoiceRecognizer(int options, boolean lazyInitialization) {
        this(null, ChoiceOptions.values()[options], lazyInitialization);
    }

    public ChoiceRecognizer(ChoiceOptions options, boolean lazyInitialization) {
        this(null, options, lazyInitialization);
    }

    public ChoiceRecognizer(boolean lazyInitialization) {
        this(null, ChoiceOptions.None, lazyInitialization);
    }

    public ChoiceRecognizer(int options) {
        this(null, ChoiceOptions.values()[options], true);
    }

    public ChoiceRecognizer(ChoiceOptions options) {
        this(null, options, true);
    }

    public ChoiceRecognizer() {
        this(null, ChoiceOptions.None, true);
    }

    public BooleanModel getBooleanModel(String culture, boolean fallbackToDefaultCulture) {
        return getModel(BooleanModel.class, culture, fallbackToDefaultCulture);
    }

    public static List<ModelResult> recognizeBoolean(String query, String culture, ChoiceOptions options, boolean fallbackToDefaultCulture) {
        
        ChoiceRecognizer recognizer = new ChoiceRecognizer(options);
        IModel model = recognizer.getBooleanModel(culture, fallbackToDefaultCulture);

        return model.parse(query);
    }

    public static List<ModelResult> recognizeBoolean(String query, String culture, ChoiceOptions options) {
        return recognizeBoolean(query, culture, options, true);
    }

    public static List<ModelResult> recognizeBoolean(String query, String culture) {
        return recognizeBoolean(query, culture, ChoiceOptions.None);
    }

    @Override
    protected void initializeConfiguration() {

        //English
        registerModel(BooleanModel.class, Culture.English, (options) -> new BooleanModel(new BooleanParser(), new BooleanExtractor(new EnglishBooleanExtractorConfiguration())));
    }
}
